hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INode.java

```diff
@@ -73,7 +73,8 @@ public abstract class INode implements INodeAttributes, Diff.Element<byte[]> {
    * Check whether this is the root inode.
    */
   final boolean isRoot() {
-    return getLocalNameBytes().length == 0;
+    // Note: There is no restriction that id must be equal to INodeId#ROOT_INODE_ID
+    return getLocalNameBytes() == null || getLocalNameBytes().length == 0;
```

Comment:
Sorry I did not get why change here. IIUC, `ROOT_NAME` is not null but empty string, right? If that, I think it will not throw NPE here. So any information I missed? Thanks @jianghuazhu .

Reply:
@Hexiaoqiao , thanks for your comment and review.
I found this question while exploring the FGL module.
In the FGL module, INodeMap#INodeMap() runs the name is null when constructing the INodeDirectory. Here: https://github.com/apache/hadoop/blob/fgl/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INodeMap.java
![image](https://user-images.githubusercontent.com/6416939/186145767-9b51ec6c-2c7b-46b0-913d-8d0385488507.png)

When verifying the lease, INode#isRoot() is called. See here:
https://github.com/apache/hadoop/blob/fgl/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java
Calling link: FSNamesystem#checkLease()->isFileDeleted()->INode#isRoot().
![image](https://user-images.githubusercontent.com/6416939/186145904-bffe4dee-a708-4f29-b728-4780a6589e8e.png)

I think this is a hidden problem because we didn't allow INodeWithAdditionalFields#name to always be an empty string.

Reply:
Got it. If that is true, I think we should submit PR to branch FGL first? Thanks.

Reply:
Thanks @Hexiaoqiao .